<?xml version="1.0"?>
<!--
config_realsense.launch
Author: Ravi Joshi
Note: Do not run this file directly. Please call run.launch file instead.
Date: 2019/10/08
-->
<launch>
  <!-- rostopic to subscribe for color images -->
  <arg name="color_topic" default="/camera/color/image_raw" />

  <!-- rostopic to subscribe for depth images -->
  <arg name="depth_topic" default="/camera/aligned_depth_to_color/image_raw" />

  <!-- rostopic to subscribe for camera calibration parameters -->
  <arg name="cam_info_topic" default="/camera/color/camera_info" />

  <!-- frame id for point cloud and 3D skeleton data -->
  <arg name="frame_id" default="camera_color_optical_frame" />

  <!-- realsense camera provide depth information.
  hence unset the flag. -->
  <arg name="no_depth" default="false" />

  <!-- note that the following parameters are going to assigned by other launch file upon call. -->
  <arg name="rviz" default="" />
  <arg name="print" default="" />
  <arg name="skeleton" default="" />
  <arg name="pub_topic" default="" />
  <arg name="pointcloud" default="" />
  <arg name="id_text_size" default="" />
  <arg name="openpose_args" default="" />
  <arg name="id_text_offset" default="" />
  <arg name="skeleton_hands" default="" />
  <arg name="skeleton_line_width" default="" />
  <arg name="synchronous" default="" />
  <arg name="py_openpose_path" default="" />

  <include file="$(find ros_openpose)/launch/core.launch" >
    <arg name="print" value="$(arg print)" />
    <arg name="no_depth" value="$(arg no_depth)" />
    <arg name="skeleton" value="$(arg skeleton)" />
    <arg name="frame_id" value="$(arg frame_id)" />
    <arg name="pub_topic" value="$(arg pub_topic)" />
    <arg name="color_topic" value="$(arg color_topic)" />
    <arg name="depth_topic" value="$(arg depth_topic)" />
    <arg name="id_text_size" value="$(arg id_text_size)" />
    <arg name="openpose_args" value="$(arg openpose_args)" />
    <arg name="id_text_offset" value="$(arg id_text_offset)" />
    <arg name="skeleton_hands" value="$(arg skeleton_hands)" />
    <arg name="cam_info_topic" value="$(arg cam_info_topic)" />
    <arg name="skeleton_line_width" value="$(arg skeleton_line_width)" />
    <arg name="synchronous" value="$(arg synchronous)" />
    <arg name="py_openpose_path" value="$(arg py_openpose_path)" />
  </include>

  <!-- invoke realsense camera ros package -->
  <group if="$(arg pointcloud)" >
    <include file="$(find realsense2_camera)/launch/rs_camera.launch" >
      <arg name="align_depth" value="true" />
      <arg name="filters" value="pointcloud" />
    </include>
    <!-- invoke rviz -->
    <group if="$(arg rviz)" >
      <node pkg="rviz" type="rviz" respawn="false" required="true" name="rviz" args="-d $(find ros_openpose)/files/person_pointcloud_realsense.rviz" />
    </group>
  </group>

  <!-- invoke realsense camera ros package -->
  <group unless="$(arg pointcloud)" >
    <include file="$(find realsense2_camera)/launch/rs_camera.launch" >
      <arg name="align_depth" value="true" />
    </include>
    <!-- invoke rviz -->
    <group if="$(arg rviz)" >
      <node pkg="rviz" type="rviz" respawn="false" required="true" name="rviz" args="-d $(find ros_openpose)/files/only_person_realsense.rviz" />
    </group>
  </group>
</launch>
